### PR TITLE
refactor: migrate task logs [DET-4826, DET-6068]

### PR DESCRIPTION
**Pull request by hkang1**
## Description

[Original Draft PR](https://github.com/determined-ai/determined/pull/3256)

NOTICE: This PR will land after [this PR lands first](https://github.com/determined-ai/determined/pull/3070).

- update task log api endpoints to be the new `api/v1/tasks/<id>/logs` endpoint
- update all logs (master, trial and task) to use the same `LogViewer` (formerly `LogViewerCore`)
- add support for task log filtering
- add empty state for log viewer
- update `LogViewer` storybook
- remove supporting and setup code for old api endpoints
- remove axios

Screen Shot 2021-12-02 at 12 35 34 PM

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by hamidzr** [comment]
I'm working on the review. there is a small build issue

```
Failed to compile.

/home/hmd/projects/da/determined/webui/react/src/components/LogViewer/LogViewer.stories.tsx
TypeScript error in /home/hmd/projects/da/determined/webui/react/src/components/LogViewer/LogViewer.stories.tsx(90,16):
Cannot find name 'jsonToMasterLog'. Did you mean 'jsonToClusterLog'?  TS2552

    88 |   <div style={{ height: '100vh' }}>
    89 |     <LogViewer
  > 90 |       decoder={jsonToClusterLog}
       |                ^
    91 |       initialLogs={[
    92 |         { id: '0', message: 'example of logs with ANSI color codes', time: '2020-06-02T21:48:07.456381-06:00' },
    93 |         { id: '1', message: ansiText, time: '2020-06-02T21:48:08.456381-06:00' },
```

**Review by hamidzr — COMMENTED** [review]
overall looks good to me love to see all those LOC deleted. 

I found a small issue: when the page loads slower than usual, it ends up with two agent filters, one of which goes away quickly
image

**Comment by hkang1** [comment]
> I'm working on the review. there is a small build issue
> 
> ```
> Failed to compile.
> 
> /home/hmd/projects/da/determined/webui/react/src/components/LogViewer/LogViewer.stories.tsx
> TypeScript error in /home/hmd/projects/da/determined/webui/react/src/components/LogViewer/LogViewer.stories.tsx(90,16):
> Cannot find name 'jsonToMasterLog'. Did you mean 'jsonToClusterLog'?  TS2552
> 
>     88 |   <div style={{ height: '100vh' }}>
>     89 |     <LogViewer
>   > 90 |       decoder={jsonToClusterLog}
>        |                ^
>     91 |       initialLogs={[
>     92 |         { id: '0', message: 'example of logs with ANSI color codes', time: '2020-06-02T21:48:07.456381-06:00' },
>     93 |         { id: '1', message: ansiText, time: '2020-06-02T21:48:08.456381-06:00' },
> ```

Doh, forgot to fix the 2nd one down below during rebase! Thanks for catching this!

File: webui/react/src/pages/TaskLogs.tsx [hunk 1]
```diff
@@ -1,108 +1,144 @@
...
 
 interface Params {
   taskId: string;
   taskType: string;
 }
 
-interface Queries {
-  id?: string;
-}
+type OrderBy = 'ORDER_BY_UNSPECIFIED' | 'ORDER_BY_ASC' | 'ORDER_BY_DESC';
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/pages/TaskLogs.tsx:24]: ah good call, we should probably use `V1OrderBy`

File: webui/react/src/pages/TaskLogFilters.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+const { Option } = Select;
+
+interface Props {
+  onChange?: (filters: Filters) => void;
+  onReset?: () => void;
+  options: Filters;
+  values: Filters;
+}
+
+export enum LogLevelFromApi {
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/pages/TaskLogFilters.tsx:16]: Yes, `V1LogLevel` makes way more sense!

**Comment by hamidzr** [comment]
please assign me again when it's good time for a rereview

**Pull request closed without merging**